### ISSUE TITLE
TryOptionAsync.Extensions.Append throws StackOverflowException

### ISSUE THREAD

**Issue by ovuncgursoy** [louthy/language-ext#597]
Here is a test case that demonstrates the issue:

```
[TestFixture]
public static class Playground
{
    private static readonly TryOptionAsync<int> Left
        = () => Task.FromResult(new OptionalResult<int>(Prelude.Some(1)));

    private static readonly TryOptionAsync<int> Right
        = () => Task.FromResult(new OptionalResult<int>(Prelude.Some(2)));

    [Test]
    public static void Append_should_not_throw_StackOverflowException()
        => Assert.DoesNotThrowAsync(() => Left.Append<TInt, int>(Right)());
}
```

I think the issue is the `this` keyword in the signature of the Append method defined in TryOptionAsync.Prelude.cs:

```
/// <summary>
/// Append the bound value of TryOptionAsync(x) to TryOptionAsync(y).  If either of the
/// Trys are Fail then the result is Fail
/// </summary>
/// <param name="lhs">Left-hand side of the operation</param>
/// <param name="rhs">Right-hand side of the operation</param>
/// <returns>lhs ++ rhs</returns>
[Pure]
public static TryOptionAsync<A> Append<SEMI, A>(this TryOptionAsync<A> lhs, TryOptionAsync<A> rhs) where SEMI : struct, Semigroup<A> =>
    lhs.Append<SEMI, A>(rhs);
```

Removing the `this` keyword makes the test pass. I think, instead of the TryOptionAsync.Extensions.Append method, at runtime, TryOptionAsync.Prelude.Append method is called.

As a side note, shouldn't TryOptionAsync.Prelude.Append be lowercase?

**Comment by gwintering** [comment]
Yep, the `Prelude` method is calling itself as an extension instead of the real extension method.  Removing `this` and lowecasing the name should fix it.

**Comment by louthy** [comment]
Fixed in `3.2.0-beta`
